appinventor/components/src/com/google/appinventor/components/runtime/Ev3ColorSensor.java

```diff
@@ -383,79 +374,80 @@ private String toColorName(String functionName, int colorCode) {
    * Specifies the mode of the sensor.
    */
   @DesignerProperty(editorType = PropertyTypeConstants.PROPERTY_TYPE_LEGO_EV3_COLOR_SENSOR_MODE,
-                    defaultValue = DEFAULT_SENSOR_MODE_STRING)
+                    defaultValue = "reflected")
   @SimpleProperty
-  public void Mode(String modeName) {
-    String functionName = "Mode";
-    try {
-      setMode(modeName);
-    } catch(IllegalArgumentException e) {
-      form.dispatchErrorOccurredEvent(this, functionName, ErrorMessages.ERROR_EV3_ILLEGAL_ARGUMENT, functionName);
+  public void Mode(@Options(ColorSensorMode.class) String modeName) {
+    // Make sure modeName is a valid ColorSensorMode.
+    ColorSensorMode mode = ColorSensorMode.fromUnderlyingValue(modeName);
+    if (mode == null) {
+      form.dispatchErrorOccurredEvent(
```

Comment:
Need to return after dispatching the error or put `setMode(mode)` in an `else` clause.